### DISCUSSION TITLE
Switched Gas tariff (from Flexible to Tracker) and have lost sensor readings

### DISCUSSION THREAD

**Discussion by ajmorr-is** [BottlecapDave/HomeAssistant-OctopusEnergy#153] [General]
As the title, I switched from Octopus Flexible to Octopus Tracker on 13th Feb and ever since then I have lost;
- current_standing_charge; appears as 'Unknown'.
- previous_accumulative_cost; appears as the same value every day and reports each day at the daily cost on the day before I switched tariff.

The variable rate works fine, and each day I get a new value for this. All other sensors relating to consumption seem OK. The Electrical sensors and figures are all still working fine.

Is this maybe a peculiarity of Octopus API when using a variable tariff (although I'm on Intelligent for electricity and that seems to work OK with the half hourly rate changes), or do I need to do something to kick it back into life?

**Comment by BottlecapDave** [comment]
There has been issues around tracker tariffs due to them requiring an undocumented API. What is the tariff code for your gas sensor? This can be found by looking for your "gas meter" device within Home Assistant and downloading the diagnostics.

**Comment by ajmorr-is** [comment]
The tariff code is "G-1R-SILVER-FLEX-22-11-25-G".

I updated the integration to v6.0.0 and the 'current_standing_charge' is now back and showing correctly.

ETA; just spotted that the previous_accumulative_cost is also back working as well. Excellent!

**Comment by ajmorr-is** [comment]
@BottlecapDave The readings and costs have all 'frozen' again for me, this time for both Electricity and Gas.
![224957978-18ea36d4-051b-422a-9bd1-085a9f335087](https://user-images.githubusercontent.com/95215735/224962333-62dfc98f-edc1-486a-8ec2-ffacf8325aaf.png)

**Reply by BottlecapDave** [reply]
Do you have any errors in your logs that would account for this? Have you upgraded the integration or HA recently?

**Reply by ajmorr-is** [reply]
Yep, plenty of errors in the log; 
![image](https://user-images.githubusercontent.com/95215735/224969980-6f2ed2fe-4e75-4096-8c99-97e10f87e680.png)

I'm running 2023.3.2 which I updated to on 9th March. The Electricity figures appear to have frozen two days ago (last correct figure received 12th March) and the Gas figures appear to have frozen three days ago (last correct figure received 11th March).

**Reply by BottlecapDave** [reply]
I can't quite see what the errors are, but..

* the first one is an error communicating with the API - this appears sometimes (I think when they do a deployment)
* The "failure to send request" errors, depending on the status code (e.g. 404), could be issues with internet connection or the product disappearing

For the dates that you mention are "frozen". The consumption and costs will probably only appear in the evening as this seems to be when OE retrieves the data. It could be they were delayed in retrieving this data (e.g. after midnight) and so it passes the threshold of when the integration retrieves the data. I would also ask if the dates you mention are present in your OE dashboard?
